app/components/add_course_button.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+class AddCourseButton < ViewComponent::Base
+  include RecruitmentCycleHelper
+  attr_reader :provider
+
+  Section = Struct.new(:name, :path)
```

Comment:
I'd add `keyword_init: true` so in the code becomes better to understand when you initialize `Section.new(name: 'aaa', path: 'bbb')

Reply:
Good idea, thanks! I definitely knew you could do that 😅 😉 

https://github.com/DFE-Digital/publish-teacher-training/pull/3697/commits/557ed4a11b9700b21c089a83714007485a61f2c8